vllm/v1/engine/processor.py

```diff
@@ -394,7 +394,7 @@ def _validate_model_input(
         else:
             tokenizer = self.tokenizer.get_lora_tokenizer(lora_request)
             max_input_id = max(prompt_ids, default=0)
-            if max_input_id > tokenizer.max_token_id:
+            if max_input_id > max(tokenizer.max_token_id, self.model_config.get_vocab_size()):
```

Comment:
perhaps no need max? model config vocab size is the source of truth.
```suggestion
            if max_input_id > self.model_config.get_vocab_size() - 1:
```

Reply:
Do we observe that we have such ids generated in Qwen?

Reply:
Yes, we have such ID generated by Qwen which is out of the tokenizer's vocabulary.

Reply:
What does that mean then? Just curious how the model / engine handle it?

Reply:
Some explanation here: https://github.com/vllm-project/vllm/issues/13175#issuecomment-2781842571
and here why vocab size is 151936(%128==0): https://github.com/QwenLM/Qwen3/issues/147#issuecomment-1988398254

Reply:
@22quinn This suggestion wouldn't work with some multimodal models that use a token in the vocabulary of the tokenizer as placeholder token but such token is included from the model vocabulary (to prevent model to generate such placeholder token itself), so I think we still need `max` here since this is for input processing.

Reply:
so in that case `tokenizer.max_token_id > self.model_config.get_vocab_size()`? Do you have an example?